egs/wsj/s5/steps/cleanup/combine_short_segments.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+
+# Copyright 2015  Tom Ko
+# Apache 2.0
+
+# Begin configuration section.
+frame_shift=10
+# End configuration section.
+
+echo "$0 $@"
+
+[ -f ./path.sh ] && . ./path.sh
+. parse_options.sh || exit 1;
+
+if [ $# -ne 3 ]; then
+  echo "This script concatenates segments with length shorter than the specified length in seconds."
+  echo ""
+  echo "Usage: steps/cleanup/combine_short_segments.sh [options] <minimum-length> <input-data-dir> <output-data-dir>"
+  echo "e.g.:  steps/cleanup/combine_short_segments.sh 2.0 data/train data/train_min2sec"
+  echo ""
+  echo "Options:"
+  echo "    --frame-shift   <frame-shift|10>         # Frame shift in milliseconds.  Only relevant if"
+  echo "                                             # <data-in>/segments does not exist (in which case"
+  echo "                                             # the lengths are to be worked out from the features)."
+  exit 1;
+fi
+
+#This script works in the unit of frames
+frame_per_sec=$(( 1000/$frame_shift ))
+min_seg_len=$(echo $1*$frame_per_sec | bc)
+input_dir=$2
+output_dir=$3
+
+for f in spk2utt text utt2spk feats.scp; do
+  [ ! -f $input_dir/$f ] && echo "$0: no such file $input_dir/$f" && exit 1;
+done
+
+export LC_ALL=C
+
+# This is the main function, to look for segments with length shorter than the specified length
+# and concatenates them with other segments to make sure every combined segments are 
+# with enough length. Here, after the search of the under-length segment, it looks for
+# another segment where the combined length is the closest to the specified length.
+
```

Comment:
I don't think it makes sense to have a function that's only called once.  In C/C++ it would be good coding practice, but it's not really traditional in bash.